### PR TITLE
test: add 39 E2E tests for fix application pipeline [skip changelog]

### LINKED ISSUE #326: test: add end-to-end fix application and edge case tests
## Context
Test coverage reviewer findings from PR #322 review.

### 1. End-to-end fix application tests
Current tests verify that diagnostics have fixes attached, but don't test the full pipeline:
- Apply fix to content -> re-validate -> zero diagnostics
- This catches bugs where fix byte ranges are slightly off

### 2. Edge case tests for YAML/JSON helpers
- CRLF line endings (`\r\n`) in frontmatter
- Unicode characters in values (multi-byte UTF-8)
- Escaped quotes inside quoted YAML values
- Very long lines (>4096 chars)
- Empty/whitespace-only values
- Nested YAML structures

### 3. Fix ordering integration test
- Multiple fixes on the same file applied in descending byte order
- Overlapping fix ranges (should be rejected or merged)

### 4. Fix safety flag tests
- Verify all `safe=true` fixes are deterministic (same input -> same output)
- Verify `safe=false` fixes are correctly marked in diagnostics

## Proposed Approach
Add a test helper:
```rust
fn assert_fix_resolves(validator: &impl Validator, path: &Path, content: &str) {
    let diags = validator.validate(path, content, &config);
    for d in &diags {
        if let Some(fix) = &d.fix {
            let fixed = apply_fix(content, fix);
            let re_diags = validator.validate(path, &fixed, &config);
            assert!(re_diags.iter().all(|d2| d2.code != d.code));
        }
    }
}
```

## Labels
enhancement, testing

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary
- Add 39 new tests covering the full fix application pipeline (validate -> apply fix -> re-validate -> zero diagnostics)
- Add YAML/JSON edge case tests: CRLF line endings, Unicode values, escaped quotes, long lines, empty values, nested YAML, mixed line endings
- Add fix ordering tests: descending byte order, overlapping range rejection, adjacent boundary fixes, same-position insertions
- Add fix safety flag tests: deterministic safe fixes, unsafe fix marking, safe-only filtering, Fix helper method correctness

## Test plan
- [x] `cargo test --package agnix-core` passes (1844 tests, 0 failures)
- [x] `cargo check` clean
- [x] `cargo clippy -D warnings` clean
- [x] No production code changes - purely additive test code

## Changes
| File | Lines | Description |
|------|-------|-------------|
| `crates/agnix-core/tests/fix_integration.rs` | +610 | NEW: 21 E2E pipeline, safety, and ordering integration tests |
| `crates/agnix-core/src/fixes.rs` | +180 | 10 fix engine edge case tests (unicode, CRLF, overlapping) |
| `crates/agnix-core/src/parsers/frontmatter.rs` | +101 | 8 frontmatter parser edge case tests |

Closes #326

### CODE REVIEW

File: crates/agnix-core/src/fixes.rs [hunk 1]
```diff
@@ -550,4 +550,184 @@ mod tests {
...
+        let fix_refs: Vec<&Fix> = fixes.iter().collect();
+
+        let (result, applied) = apply_fixes_to_content(content, &fix_refs);
+
+        assert_eq!(applied.len(), 2, "Same-position insertions should both apply");
+        // Both insertions at position 5: first "!" is inserted, then "?" is inserted
+        // at the same position in the original content (but string has shifted)
+        assert!(result.contains("!"), "First insertion should be present");
+        assert!(result.contains("?"), "Second insertion should be present");
+    }
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/fixes.rs:647]: Addressed in ea5f8a1. Added explicit sorting with sort_by, replaced weak assertions with specific assert_eq!(result, "hello?!"), and updated comments to explain stable sort behavior. Thank you for the thorough review!

File: crates/agnix-core/tests/fix_integration.rs [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+use std::path::Path;
+use std::sync::Arc;
+
+// ============================================================================
+// Shared Helper
+// ============================================================================
+
+/// Validate content, apply all fixes for `target_rule`, re-validate, and assert the
+/// target rule no longer fires.
+fn assert_fix_resolves(file_type: FileType, path: &Path, content: &str, target_rule: &str) {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/tests/fix_integration.rs]: Addressed in ea5f8a1. Updated the doc comment to clarify this is a rule-specific fix test, not a full resolution test. Other rules may still fire after fixing the target rule.

File: crates/agnix-core/src/parsers/frontmatter.rs [hunk 3]
```diff
@@ -178,6 +178,107 @@ Body content here"#;
...
+        assert!(parts.body_start <= content.len());
+
+        // frontmatter_start is at byte 3 (after "---")
+        assert_eq!(parts.frontmatter_start, 3);
+
+        // The frontmatter string is from after "---" to where "\n---" is found.
+        // Content after "---": "\r\nname: test\r\n---\r\nbody"
+        // find("\n---") matches at position where \n is the \n in \r\n before ---
+        // The match is at index 14 in rest ("\r\nname: test\r" = 14 chars, then "\n---")
+        // So frontmatter = rest[..14] = "\r\nname: test\r"
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/parsers/frontmatter.rs:212]: Addressed in ea5f8a1. Fixed the comment to use correct byte index (13 instead of 14) and clarified it's measuring bytes not characters.
